### DISCUSSION TITLE
Toggle staged and non staged files

### DISCUSSION THREAD

**Discussion by fiddej** [gitextensions/gitextensions#9627] [Ideas] [answered]
I am missing the feature to toggle all changes in staged and non staged when about to make a new commit. Right now it is a "Stage" and a "Unstage" button. But if you want to change places of files in each section, then it is unnecessarily complicated.

Use case: 
Image you have made a commit. Then you start working on something new. You then have some local changes, but then remember that you would like to amend your previous commit. If there was a toggle button, then you could stage your current changes, do what you forgot and then press the toggle button and amend you commit. 

Now it is a bit more complicated, where you have to commit your current changes, create a new branch, check out that one, do your changes, and then and either rebase or cherry pick you commit. 

![toggleButtonGitExtensions](https://user-images.githubusercontent.com/1898217/136536453-d5ca11a5-f68b-44af-85e3-63b0318b6be5.png)

**Comment by pmiossec** [comment, marked as answer]
That's seems a very convenient feature to have from a user point of view **but** except if you give us the git command that do that, I don't think that it's something that GitExtensions should do, IMHO.

Because there are so much way these commands could fail and let the user in a strange state...

You'd better create your own script to do that...

**Comment by mstv** [comment]
I would use this feature, too. But it can be even changes in the same files. I agree with @pmiossec, this should be implemented in git.
So, rather **stage** the changes you want to postpone before you change the things to be amended.
